monkey/tests/unit_tests/infection_monkey/island_api_client/test_http_client.py

```diff
@@ -0,0 +1,119 @@
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+import requests
+import requests_mock
+from urllib3.exceptions import ConnectTimeoutError
+
+from common import AgentRegistrationData
+from common.types import SocketAddress
+from infection_monkey.island_api_client import (
+    IslandAPIConnectionError,
+    IslandAPIError,
+    IslandAPIRequestError,
+    IslandAPIRequestFailedError,
+    IslandAPITimeoutError,
+)
+from infection_monkey.island_api_client.http_client import RETRIES, HTTPClient
+
+SERVER = SocketAddress(ip="1.1.1.1", port=9999)
+WINDOWS = "windows"
+AGENT_ID = UUID("80988359-a1cd-42a2-9b47-5b94b37cd673")
+AGENT_REGISTRATION = AgentRegistrationData(
+    id=AGENT_ID,
+    machine_hardware_id=1,
+    start_time=0,
+    parent_id=None,
+    cc_server=SERVER,
+    network_interfaces=[],
+)
+
+TIMESTAMP = 123456789
+
+ISLAND_URI = f"https://{SERVER}/api?action=is-up"
+LOG_ENDPOINT = f"agent-logs/{AGENT_ID}"
+ISLAND_SEND_LOG_URI = f"https://{SERVER}/api/{LOG_ENDPOINT}"
+ISLAND_GET_AGENT_BINARY_URI = f"https://{SERVER}/api/agent-binaries/{WINDOWS}"
+ISLAND_SEND_EVENTS_URI = f"https://{SERVER}/api/agent-events"
+ISLAND_REGISTER_AGENT_URI = f"https://{SERVER}/api/agents"
+ISLAND_AGENT_STOP_URI = f"https://{SERVER}/api/monkey-control/needs-to-stop/{AGENT_ID}"
+ISLAND_GET_CONFIG_URI = f"https://{SERVER}/api/agent-configuration"
+PROPAGATION_CREDENTIALS_ENDPOINT = "propagation-credentials"
+ISLAND_GET_PROPAGATION_CREDENTIALS_URI = f"https://{SERVER}/api/{PROPAGATION_CREDENTIALS_ENDPOINT}"
+ISLAND_GET_AGENT_SIGNALS = f"https://{SERVER}/api/agent-signals/{AGENT_ID}"
+
+
+@pytest.fixture
+def connected_client():
+    http_client = HTTPClient()
+
+    with requests_mock.Mocker() as m:
+        m.get(ISLAND_URI)
+        http_client.connect(SERVER)
+        yield {"client": http_client, "mock": m}
```

Comment:
I think these can be two fixtures, which would significantly improve readability:

```python
import pytest
import requests
import requests_mock

@pytest.fixture
def requests_mock_instance():
    with requests_mock.Mocker() as m:
        print("yielding")
        yield m
        print("cleaning up")

@pytest.fixture
def other_component(requests_mock_instance):
    print("constructing other_component")
    return None


def test_stuff(requests_mock_instance, other_component):
    print("running test")
    assert False
```

```
tests/unit_tests/infection_monkey/test_mike.py::test_stuff FAILED                                                                                       [100%]

========================================================================== FAILURES ===========================================================================
_________________________________________________________________________ test_stuff __________________________________________________________________________

requests_mock_instance = <requests_mock.mocker.Mocker object at 0x7fb544ddfac0>, other_component = None

    def test_stuff(requests_mock_instance, other_component):
        print("running test")
>       assert False
E       assert False

tests/unit_tests/infection_monkey/test_mike.py:20: AssertionError
-------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------
yielding
constructing other_component
-------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------
running test
------------------------------------------------------------------ Captured stdout teardown -------------------------------------------------------------------
cleaning up
```